velox/exec/LocalPlanner.cpp

```diff
@@ -257,14 +257,47 @@ void LocalPlanner::plan(
 
   (*driverFactories)[0]->outputDriver = true;
 
+  // Determine which pipelines should run Grouped Execution.
+  if (planFragment.isGroupedExecution()) {
+    // See if pipelines have leaf nodes that use grouped execution strategy.
+    for (auto& factory : *driverFactories) {
+      if (planFragment.leafNodeRunsGroupedExecution(
+              factory->planNodes.front()->id())) {
+        factory->groupedExecution = true;
+      }
+    }
+
+    // If all sources of a pipeline belong pipelines that run Grouped
```

Comment:
This doesn't sound quite right. The rule applies only to LocalExchange and it says "if all sources run grouped, then pipeline runs grouped". For example, this rule does not apply to a join.

Reply:
Will replace by
```
    // If a pipeline's leaf node is Local Partition, which has all sources
    // belonging to pipelines that run Grouped Execution, then our pipeline
    // should run Grouped Execution as well.
```